libraries/AP_HAL_PX4/Scheduler.h

```diff
@@ -13,7 +13,7 @@
 #define APM_MAIN_PRIORITY_BOOST 241
 #define APM_MAIN_PRIORITY       180
 #define APM_TIMER_PRIORITY      181
-#define APM_SPI_PRIORITY        182
+#define APM_SPI_PRIORITY        242
```

Comment:
In other words, you always guarantee main thread has CPU available for completing its job, and have it to wait and boost the priority of whatever it's waiting for.